### PR TITLE
Fixes #3320 - Review Jetty 10 module-info.java.

### LINKED ISSUE #3320: Review Jetty 10 module-info.java
`module-info.java` files are there for Jetty 10.

We need to review them to make sure that the dependencies are correct (e.g. that `requires static` or `require transitive` are correct).
Order packages in alphabetical order.

### PR DESCRIPTION
**Pull request by sbordet**
Issue #3320.

Reviewed module-info.java files.
For those that have a "requires static" added a comment about
why the dependency is optional.
Rearranged directives in alphabetical order.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

### CODE REVIEW

**Comment by sbordet** [comment]
@joakime, most notable changes are:

* in `fcgi-server` module, made dependency to `jetty-servlet-api` optional (and correspondent change in `pom.xml`).
* in `jetty-util` module, removed dependency to `jetty-servlet-api` (as the multipart classes have been moved to `jetty-http`).

The rest is just shuffling around to make things sorted alphabetically.

File: jetty-plus/src/main/java/module-info.java [hunk 1]
```diff
@@ -29,14 +29,19 @@
...
     requires org.eclipse.jetty.webapp;
     requires org.eclipse.jetty.xml;
+
+    // Only required if using DataSourceLoginService.
     requires static java.sql;
+    // Only required if using ContainerInitializer.
     requires static jetty.servlet.api;
+    // Only required if using DataSourceLoginService.
     requires static org.eclipse.jetty.server;
+    // Only required if using RunAs.
```
Comment (joakime) [inline jetty-plus/src/main/java/module-info.java]: I believe there is no valid use case for using jetty-plus without the servlet spec.
jndi and initial context were mentioned, but those can be used without jetty-plus as well, just relying on the default implementation in the JVM.
the jndi and initial context provided by jetty-plus are for a servlet specific behaviors and requirements.

File: jetty-plus/src/main/java/module-info.java [hunk 2]
```diff
@@ -29,14 +29,19 @@
...
     requires java.naming;
     requires java.transaction;
-    requires org.eclipse.jetty.util;
     requires org.eclipse.jetty.jndi;
     requires org.eclipse.jetty.security;
+    requires org.eclipse.jetty.util;
     requires org.eclipse.jetty.webapp;
     requires org.eclipse.jetty.xml;
+
+    // Only required if using DataSourceLoginService.
```
Comment (joakime) [inline jetty-plus/src/main/java/module-info.java]: Can you use the LoginService without the servlet api?

Comment (joakime) [reply jetty-plus/src/main/java/module-info.java]: Nope, you need the servlet-api

File: jetty-util/src/main/java/module-info.java [hunk 3]
```diff
@@ -32,12 +32,17 @@
...
+    // Only required if using X509.
     requires static java.naming;
+    // Only required if using JavaUtilLog.
     requires static java.logging;
+    // Only required if using DriverManagerLeakPreventer.
     requires static java.sql;
+    // Only required if using DOMLeakPreventer.
     requires static java.xml;
-    requires static jetty.servlet.api;
+    // Only required when using Jetty logging.
```
Comment (joakime) [inline jetty-util/src/main/java/module-info.java]: // Only required when using Slf4jLog

**Pull request merged**